### PR TITLE
mon: osd pool create <pool-name> with --bulk flag

### PR DESCRIPTION
**Pull request by kamoltat**
Creating the pool with `--bulk` will allow
the pg_autoscaler to use the `scale-down`
mode on.

Creating pool:

`ceph osd pool create <pool-name> --bulk`

Get var:

`ceph osd pool get <pool-name> bulk`

Set var:

`ceph osd pool set <pool-name> bulk=true/false/1/0`

To do:

- [x]  incorporate the bulk flag to autoscale behavior
- [x] hide autoscale profiles from users and use bulk instead
- [x] modify existing test cases that used autoscale profile
- [x] add more test cases for bulk flag 

Signed-off-by: Kamoltat <ksirivad@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by kamoltat** [comment]
https://pulpito.ceph.com/ksirivad-2021-12-15_06:42:55-rados:singleton:all:pg-autoscaler.yaml-wip-ksirivad-pool-bulk-flag-distro-basic-smithi/6563716/

File: src/pybind/mgr/pg_autoscaler/module.py [hunk 1]
```diff
@@ -567,15 +543,15 @@ def _calc_pool_targets(
                 'pg_num_final': final_pg_target,
                 'would_adjust': adjust,
                 'bias': p.get('options', {}).get('pg_autoscale_bias', 1.0),
+                'bulk': bulk,
             })
 
-        return ret, even_pools
+        return ret, bulk_pools, even_pools
```
[the remark this answers is not in the document]
Comment (kamoltat) [reply src/pybind/mgr/pg_autoscaler/module.py:549]: Fixed! thank you

**Comment by kamoltat** [comment]
jenkins test make check

**Comment by kamoltat** [comment]
Work Unit r`ados:singleton:all:pg-autoscaler.yaml` Passed 1/1

https://pulpito.ceph.com/ksirivad-2021-12-17_02:07:07-rados:singleton:all:pg-autoscaler.yaml-wip-ksirivad-pool-bulk-flag-distro-basic-smithi/6566864/

**Review by jdurgin — APPROVED** [review]

**Comment by kamoltat** [comment]
Tested the qa/workunits/cephtool/test.sh: 1/1 passed
https://pulpito.ceph.com/ksirivad-2021-12-20_18:43:43-rados:singleton-bluestore:all:cephtool.yaml-wip-ksirivad-pool-bulk-flag-distro-basic-smithi/6573367/

**Comment by neha-ojha** [comment]
> Tested the qa/workunits/cephtool/test.sh: 1/1 passed https://pulpito.ceph.com/ksirivad-2021-12-20_18:43:43-rados:singleton-bluestore:all:cephtool.yaml-wip-ksirivad-pool-bulk-flag-distro-basic-smithi/6573367/

@kamoltat let's also run this through a broader rados suite, @yuriw can you please pick this up for your next batch?

**Comment by yuriw** [comment]
@kamoltat ack pls rebase

**Comment by yuriw** [comment]
jenkins test api

**Comment by kamoltat** [comment]
http://pulpito.front.sepia.ceph.com/yuriw-2021-12-21_18:01:07-rados-wip-yuri3-testing-2021-12-21-0749-distro-default-smithi/

**Comment by kamoltat** [comment]
Here is the failure analysis:
https://pad.ceph.com/p/teuthology_failures

**Comment by neha-ojha** [comment]
> Here is the failure analysis: https://pad.ceph.com/p/teuthology_failures

@kamoltat I don't see mentions of the dead jobs, have you analyzed them to verify that they not related?

**Comment by kamoltat** [comment]
> > Here is the failure analysis: https://pad.ceph.com/p/teuthology_failures
> 
> @kamoltat I don't see mentions of the dead jobs, have you analyzed them to verify that they not related?

yes, looking at it right now thanks for reminding

**Comment by neha-ojha** [comment]
All failed and dead jobs in http://pulpito.front.sepia.ceph.com/yuriw-2021-12-21_18:01:07-rados-wip-yuri3-testing-2021-12-21-0749-distro-default-smithi/ are unrelated, list of unrelated trackers in https://pad.ceph.com/p/teuthology_failures

**Comment by nh2** [comment]
@kamoltat  There are some remaining references:

```
% git grep -E 'scale-down|scale-up'
src/common/options/global.yaml.in:  fmt_desc: Set the ``bulk`` flag on new pools. Allowing autoscaler to use scale-down mode.
src/pybind/mgr/pg_autoscaler/module.py:        scaling, this depends on the profile of the autoscaler. For scale-down,
src/pybind/mgr/pg_autoscaler/module.py:        pools needs more pgs due to increased usage. For scale-up, we start out with
```

Also the release notes did not really specify that `mgr/pg_autoscaler/autoscale_profile` is now unused; I noticed I still have it left-over in my config.

**Pull request merged**